uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/main.controller.js

```diff
@@ -35,7 +37,7 @@ define([
       var method = $rootScope['modal' + _.capitalize(modalType)];
 
       if (typeof method === 'function') {
-        method();
+        method({ case_type_id: (queryParams.caseTypeId || null) });
```

Comment:
Isn't it confusing to pass `{ case_type_id: null }` to all modals other than modalDocument?

Why not do:

```js
function modalDocument (data, modalMode, e) {
  // ...
  data = data || {};
  data.case_type_id = data.case_type_id || queryParams.caseTypeId || null;
  // ...
}
```

Reply:
the modal is modalAssignment, to be clear.

I initially did what you proposed but it doesn't work correctly, as it would use the query params even when you manually open the modal (by clicking on the "add workflow" button). You want the case type to be auto-selected only when the modal is opened automatically.

And yes i wasn't much too happy with passing case_type_id to all the modal types, but i don't know how to pass it only to the assignment modal without making a ad-hoc condition.

So I thought that in the end it shouldn't be a problem: if the modal uses that value, fine, otherwise it will just ignore it